### ISSUE TITLE
3.14

### ISSUE THREAD

**Issue by snowboundmage2** [dolkensp/unp4k#43]
don't think this is compatible, only gets a few files then crashes

**Comment by peter-dolkens** [comment]
Have you made sure to download the latest release?

I'm not getting any issues in my testing.

**Comment by peter-dolkens** [comment]
Also check you have space to extract the files you have requested.

If you haven't provided a filter, then you need around 120GB of space to extract the p4k file.

![image](https://user-images.githubusercontent.com/469666/131401347-179c2f1a-adb0-43b2-9d74-1e137f9228f0.png)

**Comment by Blue-0xEF** [comment]
Performed as expected.

Had no issues.

But I'm only looking to extract the controls defaultProfile.xml.

Thanks for maintaining this for me.

 

  _____

**Comment by AusKarlos87** [comment]
Im having some issues with it extracting some files as well. It seems a fresh download of 3.14 works fine, but any pack file that was a fresh download from before 3.12 (and patched all the way to 3.14) seems to crash unp4k frequently.
The cli version seems to just skip over the files where the GUI version crashes.

**Comment by peter-dolkens** [comment]
Thanks, that's really helpful

I've got a p4k archive so I'll see if I can replicate

Peter Dolkens

**Comment by sofyanet** [comment]
Hi,

I don't mean to high jack this issue ticket but I'm having issues with the GUI version. It does start analyzing the .p4k file then hangs after 98.66% completed.

![image](https://user-images.githubusercontent.com/65699547/144921050-5a3f6688-926c-44d3-af6e-9be3021e6af9.png)

I thought it might be related. I couldn't manage to fully unpack the file either using the despite having enough disk space. However, after running the GUI through VS, it appears that I hit a memory limitation:

System.OutOfMemoryException
  HResult=0x8007000E
  Message=Une exception de type 'System.OutOfMemoryException' a été levée.
  Source=PresentationCore
  Arborescence des appels de procédure :
   à System.Windows.Media.Animation.TimeManager.Tick()
   à System.Windows.Media.MediaContext.RenderMessageHandlerCore(Object resizedCompositionTarget)
   à System.Windows.Media.MediaContext.AnimatedRenderMessageHandler(Object resizedCompositionTarget)
   à System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   à System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   à System.Windows.Threading.DispatcherOperation.InvokeImpl()
   à System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(Object state)
   à MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   à System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   à System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   à System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   à MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   à System.Windows.Threading.DispatcherOperation.Invoke()
   à System.Windows.Threading.Dispatcher.ProcessQueue()
   à System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   à MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   à MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   à System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   à System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   à System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   à MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   à MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   à System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   à System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   à System.Windows.Application.RunDispatcher(Object ignore)
   à System.Windows.Application.RunInternal(Window window)
   à System.Windows.Application.Run(Window window)
   à unp4k.gui.App.Main()

![image](https://user-images.githubusercontent.com/65699547/144921513-6eb57ed5-f297-4689-aba5-3dcd8abf4a89.png)

My system runs on 16 GB + 32 GB of paging.

Hope this helps.

**Comment by sofyanet** [comment]
Hi,

I don't mean to high jack this issue ticket but I thought my problem might be related. I'm having issues with the GUI version. It does start analyzing the .p4k file then hangs after 98.66% completed.

![image](https://user-images.githubusercontent.com/65699547/144921050-5a3f6688-926c-44d3-af6e-9be3021e6af9.png)

However, I had no issue unpacking the archive using the non GUI version. After running the GUI version through debug in VS, it appears that I hit a memory limitation:

System.OutOfMemoryException
  HResult=0x8007000E
  Message=Une exception de type 'System.OutOfMemoryException' a été levée.
  Source=PresentationCore
  Arborescence des appels de procédure :
   à System.Windows.Media.Animation.TimeManager.Tick()
   à System.Windows.Media.MediaContext.RenderMessageHandlerCore(Object resizedCompositionTarget)
   à System.Windows.Media.MediaContext.AnimatedRenderMessageHandler(Object resizedCompositionTarget)
   à System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   à System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   à System.Windows.Threading.DispatcherOperation.InvokeImpl()
   à System.Windows.Threading.DispatcherOperation.InvokeInSecurityContext(Object state)
   à MS.Internal.CulturePreservingExecutionContext.CallbackWrapper(Object obj)
   à System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   à System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state, Boolean preserveSyncCtx)
   à System.Threading.ExecutionContext.Run(ExecutionContext executionContext, ContextCallback callback, Object state)
   à MS.Internal.CulturePreservingExecutionContext.Run(CulturePreservingExecutionContext executionContext, ContextCallback callback, Object state)
   à System.Windows.Threading.DispatcherOperation.Invoke()
   à System.Windows.Threading.Dispatcher.ProcessQueue()
   à System.Windows.Threading.Dispatcher.WndProcHook(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   à MS.Win32.HwndWrapper.WndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam, Boolean& handled)
   à MS.Win32.HwndSubclass.DispatcherCallbackOperation(Object o)
   à System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   à System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)
   à System.Windows.Threading.Dispatcher.LegacyInvokeImpl(DispatcherPriority priority, TimeSpan timeout, Delegate method, Object args, Int32 numArgs)
   à MS.Win32.HwndSubclass.SubclassWndProc(IntPtr hwnd, Int32 msg, IntPtr wParam, IntPtr lParam)
   à MS.Win32.UnsafeNativeMethods.DispatchMessage(MSG& msg)
   à System.Windows.Threading.Dispatcher.PushFrameImpl(DispatcherFrame frame)
   à System.Windows.Threading.Dispatcher.PushFrame(DispatcherFrame frame)
   à System.Windows.Application.RunDispatcher(Object ignore)
   à System.Windows.Application.RunInternal(Window window)
   à System.Windows.Application.Run(Window window)
   à unp4k.gui.App.Main()

![image](https://user-images.githubusercontent.com/65699547/144921513-6eb57ed5-f297-4689-aba5-3dcd8abf4a89.png)

My system runs on 16 GB + 32 GB of paging.

Hope this helps.

**Comment by Stryxus** [comment]
@sofyanet Yes, this is most likely due to so many UI elements being created and stored into memory before actually being rendered. This will be fixed when myself and/or peter get a chance to start on the new MAUI GUI. (I rewrote a large chunk of unp4k/unforge which is just awaiting peter's approval and possible release although, the GUI in it had to be removed for the time being)

[feature/net6](https://github.com/dolkensp/unp4k/tree/feature/net6)

**Comment by sofyanet** [comment]
@Stryxus Thank you for all those information.
